Code/Framework/AzCore/AzCore/Math/Internal/SimdMathCommon_simd.inl

```diff
@@ -537,6 +537,16 @@ namespace AZ
             }
 
 
+            template <typename VecType>
+            AZ_MATH_INLINE void Mat4x4MultiplyAdd(const typename VecType::FloatType* __restrict rowsA, const typename VecType::FloatType* __restrict rowsB, typename VecType::FloatType* __restrict out)
+            {
+                out[0] = VecType::Madd(VecType::SplatFourth(rowsA[0]), rowsB[3], VecType::Madd(VecType::SplatThird(rowsA[0]), rowsB[2], VecType::Madd(VecType::SplatSecond(rowsA[0]), rowsB[1], VecType::Madd(VecType::SplatFirst(rowsA[0]), rowsB[0], out[0]))));
+                out[1] = VecType::Madd(VecType::SplatFourth(rowsA[1]), rowsB[3], VecType::Madd(VecType::SplatThird(rowsA[1]), rowsB[2], VecType::Madd(VecType::SplatSecond(rowsA[1]), rowsB[1], VecType::Madd(VecType::SplatFirst(rowsA[1]), rowsB[0], out[1]))));
+                out[2] = VecType::Madd(VecType::SplatFourth(rowsA[2]), rowsB[3], VecType::Madd(VecType::SplatThird(rowsA[2]), rowsB[2], VecType::Madd(VecType::SplatSecond(rowsA[2]), rowsB[1], VecType::Madd(VecType::SplatFirst(rowsA[2]), rowsB[0], out[2]))));
+                out[3] = VecType::Madd(VecType::SplatFourth(rowsA[3]), rowsB[3], VecType::Madd(VecType::SplatThird(rowsA[3]), rowsB[2], VecType::Madd(VecType::SplatSecond(rowsA[3]), rowsB[1], VecType::Madd(VecType::SplatFirst(rowsA[3]), rowsB[0], out[3]))));
```

Comment:
Should we have used zero based naming for the Splat functions?
Such as `SplatZeroth`, so that it matched with the 0-based indexing of C arrays.

Reply:
Yeah, that might have been better.. the zero indexed array thing comes from a very elegant idea in C.
https://medium.com/@jondsmith22/a-love-letter-to-cs-square-brackets-462e16f168ae
But it does mean that we associate index 1 with the second element in an array now.
Perhaps SplatElement0, SplatElement1, SplatElement2, and so forth?

Reply:
An alias such as `SplatElement0`, `SplatElement1` or even `SplatIndex0` could help reduce mistakes in the future.

Reply:
Also should we initialize the output vector to all `0`s otherwise, the MultiplyAndAdd the operations of 
```
Splat0th(rowA[0]) * rowB[0] + out[0]
Splat0th(rowA[1]) * rowB[0] + out[1]
Splat0th(rowA[2]) * rowB[0] + out[2]
Splat0th(rowA[3]) * rowB[0] + out[4]
``` 

Would result in indeterminate values if the output vector has not been initialized?

Finally should we have the output parameter be a `span<typename VecType::FloatType, 4>` so that we are sure the user is passing in container or pointer that can store 4 elements.

Be that a call to 
```c++
// C-style array
__m128 outCArray[4]{};
Mat4x4MultiplyAdd(rowsA, rowsB, outCArray);

// Std array class
AZStd::array<__m128, 4> outCPPArray{};
Mat4x4MultiplyAdd(rowsA, rowsB, outCPPArray);

// Pointer + size
constexpr size_t size = 4;
__m128 pointer = new __m128[size];
Mat4x4MultiplyAdd(rowsA, rowsB, AZStd::span(pointer, size));
...
```

Reply:
std::array and std::span both probably won't work in such a core low level segment of the math libs, for reasons of speed, compile-time, and simple separation of code. The c-style array declaration might be viable, since there are only a few touchpoints here. I'd like to separate that as a change though since it would be a breaking API change? Currently our Matrix4x4 constructs a set of Vector4's, not the lower-level SIMD types even though the layout in memory is completely identical, and we make pretty heavy usage of that abstraction to simplify code; so short of a big rewrite we'd just be forcibly casting things for now.